### ISSUE TITLE
url_launcher 1.0.0 crashes on iOS when dealing with non-web links

### ISSUE THREAD

**Issue by mit-mit** [flutter/flutter#13067]
## Steps to Reproduce

Run this program:

```
import 'package:flutter/material.dart';
import 'package:url_launcher/url_launcher.dart';

void main() => runApp(new MaterialApp(
      home: new MyHomePage(),
    ));

class MyHomePage extends StatelessWidget {
  void handleTap() {
    launch('mailto:smith@example.org?subject=News&body=New%20plugin');
  }

  @override
  Widget build(BuildContext context) {
    return new Scaffold(
      appBar: new AppBar(
        title: new Text('url_launcher test'),
      ),
      body: new Text('Click the button'),

      floatingActionButton: new FloatingActionButton(
        onPressed: handleTap,
        tooltip: 'Increment',
        child: new Icon(Icons.add),
      ),
    );
  }
}
```

on a physical device. Once you click the FAB it crashes.

The crash only occurs on the not yet published version `1.0.0` of the plugin, so you need to pull that in `pubspec.yaml`:

```
  url_launcher:
    git:
      url: git://github.com/flutter/plugins/
      path: packages/url_launcher
```

The crash doesn't against the published plugin, i.e.:

```
  url_launcher:
```

**Comment by mit-mit** [comment]
I think this was caused by https://github.com/flutter/flutter/issues/11584. cc @xster

**Comment by xster** [comment]
Thanks for catching. This is kinda fixed by https://github.com/flutter/plugins/pull/276

**Comment by mehmetf** [comment]
@mit-mit @xster I want to emphasize the "kinda".

Since only web content is intended for VC, should we put logic to launcher plugin that looks at the scheme and decides what to open?

This has implications on flutter/plugins#276 design. Instead of having an option flag, we would need an override flag which is null by default.

**Comment by xster** [comment]
We can probably assert and validate in dart. If VC and url not scheme = x, crash debug and ignore VC param in production.

**Comment by mehmetf** [comment]
Closed with flutter/plugins#276
